src/main/scala/zooklabs/program/UpdateLeagueProgram.scala

```diff
@@ -2,21 +2,56 @@ package zooklabs.program
 
 import cats.effect.{IO, Timer}
 import cats.implicits._
+import eu.timepit.refined.types.all.NonNegInt
+import eu.timepit.refined.types.string.NonEmptyString
 import fs2.Stream
 import zooklabs.`enum`.Trials
+import zooklabs.model.LeagueTrial
 import zooklabs.repository.LeagueRepository
+import zooklabs.repository.model.LeagueRanksContainer
 
 import scala.concurrent.duration._
 
 final class UpdateLeagueProgram(leagueRepository: LeagueRepository)(implicit timer: Timer[IO]) {
 
-  def run(): Stream[IO, Unit] = {
-    (Stream.eval(Trials.values.traverse(updateLeague(leagueRepository))) ++ Stream
-      .sleep_[IO](1.hour)).repeat.void
+  // Overall has its own scoring and ranking
+  val isStandardTrial: Trials => Boolean = _ != Trials.Overall
+
+  def updateLeague = {
+    Trials.values
+      .filter(isStandardTrial)
+      .traverse(leagueRepository.updateLeagues) >> updateOverallLeague
+  }
+
+  def getOverallScores(container: LeagueRanksContainer): List[LeagueTrial] = {
+
+    case class UnrankedTrial(id: NonNegInt, name: NonEmptyString, score: Double)
+
+    val rankTrial = (trial: UnrankedTrial, index: Int) =>
+      LeagueTrial(trial.id, trial.name, trial.score, index + 1)
+
+    val overallResults = for {
+      leagueRanks <- container.leagueRanks
+      leagueCounts = container.leagueCounts
+      overallScore = OverallScoreCalculations.calculateOverallScore(leagueRanks, leagueCounts)
+    } yield UnrankedTrial(leagueRanks.id, leagueRanks.name, overallScore)
+
+    val sortedResults = overallResults.sortBy(-_.score)
+
+    (for ((result, index) <- sortedResults.view.zipWithIndex) yield rankTrial(result, index)).toList
```

Comment:
ill see what i can do with this , i think it could be done a bit more succinctly, but not a deal breaker if it works